test/streams.js

```diff
@@ -45,3 +47,66 @@ describe('strong-remoting', function() {
       .expect({bar: 'baz'}, done);
   });
 });
+
+describe('a function returning a ReadableStream', function() {
+  var Readable = require('stream').Readable;
+  var remotes = RemoteObjects.create();
+  var streamClass;
+  var server;
+
+  before(function(done) {
+    var app = express();
+    server = app.listen(0, '127.0.0.1', done);
+    function StreamClass() {
+
+    }
+
+    StreamClass.createStream = function createStream(cb) {
+      var rs = new Readable({objectMode: true});
+      var buf = [];
+      var n = 5;
+      while (n--) {
+        buf.push({n: n});
+      }
+      rs._read = function() {
+        setTimeout(function() {
+          rs.push(buf.shift() || null);
+        }, 1);
+      };
+      cb(null, rs);
+    };
+
+    streamClass = new SharedClass('StreamClass', StreamClass);
+
+    var method = streamClass.defineMethod('createStream', {
+      isStatic: true,
+      fn: StreamClass.createStream,
+      returns: [{
+        arg: 'stream',
+        type: 'ReadableStream',
+        json: true
+      }]
+    });
+
+    remotes.addClass(streamClass);
+    app.use(remotes.handler('rest'));
+  });
+
+  before(function() {
+    remotes.connect('http://127.0.0.1:' + server.address().port, 'rest');
+  });
+
+  it('should return a ReadableStream', function(done) {
+    remotes.invoke('StreamClass.createStream', [], function(err, stream) {
+      assert(stream.readable, 'must be a ReadableStream');
+      var count = 0;
+      stream.on('data', function(d) {
+        count++;
+
+        if (count === 5) {
+          done();
+        }
+      });
```

Comment:
Perhaps use [writeArray](https://www.npmjs.com/package/event-stream#writearray-callback) and assert on the array content?